src/bp-moderation/bp-moderation-filters.php

```diff
@@ -523,3 +523,67 @@ function bb_moderation_suspend_after_delete( $recode ) {
 
 }
 add_action( 'suspend_after_delete', 'bb_moderation_suspend_after_delete' );
+
+/**
+ * Function to clear the cache data on item suspend.
+ *
+ * @since BuddyBoss 1.5.6
+ *
+ * @param array $moderation_data moderation item data.
+ */
+function bb_moderation_clear_suspend_cache( $moderation_data ) {
+	if ( empty( $moderation_data['item_type'] ) || empty( $moderation_data['item_id'] ) ) {
+		return;
+	}
+	wp_cache_delete( 'bb_check_moderation_' . $moderation_data['item_type'] . '_' . $moderation_data['item_id'], 'bb' );
+	wp_cache_delete( 'bb_check_hidden_content_' . $moderation_data['item_type'] . '_' . $moderation_data['item_id'], 'bb' );
+	wp_cache_delete( 'bb_check_suspended_content_' . $moderation_data['item_type'] . '_' . $moderation_data['item_id'], 'bb' );
+	wp_cache_delete( 'bb_check_user_suspend_' . $moderation_data['item_type'] . '_' . $moderation_data['item_id'], 'bb' );
+}
+
+add_action( 'bp_suspend_before_add_suspend', 'bb_moderation_clear_suspend_cache' );
+add_action( 'bp_suspend_before_remove_suspend', 'bb_moderation_clear_suspend_cache' );
+
+/**
+ * Function to clear cache on suspend item delete.
+ *
+ * @since BuddyBoss 1.5.6
+ *
+ * @param object $suspend_record suspend item record.
+ */
+function bp_moderation_clear_delete_cache( $suspend_record ) {
+	if ( empty( $suspend_record->item_type ) || empty( $suspend_record->item_id ) ) {
+		return;
+	}
+	wp_cache_delete( 'bb_check_moderation_' . $suspend_record->item_type . '_' . $suspend_record->item_id, 'bb' );
+	wp_cache_delete( 'bb_check_hidden_content_' . $suspend_record->item_type . '_' . $suspend_record->item_id, 'bb' );
+	wp_cache_delete( 'bb_check_suspended_content_' . $suspend_record->item_type . '_' . $suspend_record->item_id, 'bb' );
+	wp_cache_delete( 'bb_check_user_suspend_' . $suspend_record->item_type . '_' . $suspend_record->item_id, 'bb' );
+}
+
+add_action( 'bp_moderation_after_save', 'bp_moderation_clear_delete_cache' );
+add_action( 'suspend_after_delete', 'bp_moderation_clear_delete_cache' );
+add_action( 'bp_moderation_after_hide', 'bp_moderation_clear_delete_cache' );
+add_action( 'bp_moderation_after_unhide', 'bp_moderation_clear_delete_cache' );
+
+/**
+ * Function to clear cache when item hide/unhide
+ *
+ * @since BuddyBoss 1.5.6
+ *
+ * @param string $content_type content type.
+ * @param int    $content_id   content id.
+ * @param array  $args         item arguments.
+ */
+function bp_moderation_clear_status_change_cache( $content_type, $content_id, $args ) {
```

Comment:
Update the function name:

```suggestion
function bb_moderation_clear_status_change_cache( $content_type, $content_id, $args ) {
```